### PR TITLE
Module 3: Storybook, MovieTile, MovieDetails and Test Cases covered

### PR DESCRIPTION
**Pull request by debasen**
# Requirement
## Install Storybook
Install and configure [Storybook](https://storybook.js.org/) by following guide from the official website.
In the end you should have a "storybook" npm script. When running "npm run storybook" it should start Storybook and open it in your browser.
## Write Storybook stories for existing components
Create a story file for every component you created in previous modules. You should have at least one story for each of the following components (in total minimum 3):
- Counter
- SearchForm
- GenreSelect

You can write more stories to cover several states of GenreSelect or SearchForm, but this is optional.
## Implement other components
Based on [Figma prototype](https://www.figma.com/file/fKGjrOqR6nJe6LYJopGCZ8/CDP-Home-Task-%E2%80%93-React-v1?node-id=0%3A2) implement the following components:
### MovieTile
This component should render a movie tile from the list of movies.
It should take properties to receive image url, movie name, release year, and a list of relevant genres. Alternatively, you can define one component property to take an object with all movie info. Additionally, the component should receive a callback property to capture click event.
Optionally, you can implement a context menu popup that opens when a user clicks on three dots button and contains "Edit" and "Delete" menu items.

### MovieDetails
This component will render movie details when a movie is selected from the list (clicked). The details include movie poster image on the left and the rest of info on the right.
The component should take properties to receive image url, movie name, release year, rating, duration and a description. Alternatively,  you can specify a single property that accepts an object with all movie info.

### SortControl
This component will render a label "Sort by" and a select control to the right. Select should have the following options:
- Release Date
- Title

The component should take a property that specifies current selection. Additionally, it should take a callback property to handle selection changes. The callback should be called every time a user changes "Sort by" value. The new value should be passed in callback arguments.
Use Storybook when implementing every individual component. Write a story for every component you create. It will help you to check the result and play around with some interactivity before you embed components into the app.
Although, it's not required to implement high-fidelity design as per design prototype, it still makes sense to apply some styling to your components, so that you practice in styling React components and your final result looks good.

### Write tests
Cover new components with tests using jest and @testing-library/react. Verify necessary data is rendered as well as that behavior works correctly. You can write snapshot tests to cover the rendering. Although, remember that snapshot tests are fragile and will fail any time you change your component markup.

# Verification
## Story Book
image

## Movie Tile
image
## Movie Details
image
## Test Cases:
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,82 @@
+
+import React, { useState, useEffect } from 'react';
+import axios from 'axios'
 import './App.css';
```
Comment (melvinvicky46) [inline src/App.js:4]: Move styling import to after all the imports to line 12

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,82 @@
...
 import GenreSelect from './components/genreSelect/genreSelect';
 import Counter from './components/counter/counter';
+import MovieTile from './components/movieTile/movieTile';
+import MovieDetails from './components/movieDetails/movieDetails';
+import SortControl from './components/sortControl/sortControl';
 import { INITAL_COUNT, INITAL_SEARCH_TERM } from './constants';
-
+;
 function App() {
+  const [data, setData] = useState([]);
```
Comment (melvinvicky46) [inline src/App.js:14]: Please proper naming convention.

**Review by melvinvicky46 — COMMENTED** [review]
Please updated the changes

File: src/App.js (hunk 2 above)
Comment (debasen) [reply src/App.js]: Fixed it.

File: src/App.js (hunk 1 above)
Comment (debasen) [reply src/App.js]: Fixed it.

**Pull request merged**